### PR TITLE
177 cumulative can not assume args are numpy array

### PR DESCRIPTION
**Pull request by IgnaceBleukx**
Fixed wrong assumption arguments are numpy arrays

### CODE REVIEW

File: cpmpy/expressions/globalconstraints.py [hunk 1]
```diff
@@ -455,7 +455,15 @@ def decompose(self):
         return cons
 
     def value(self):
-        start, dur, end, demand, cap = [argval(a) for a in self.args]
+        import numpy as np
```
Comment (tias) [inline cpmpy/expressions/globalconstraints.py:458]: why here and not at top level?

File: cpmpy/expressions/globalconstraints.py [hunk 2]
```diff
@@ -455,7 +455,15 @@ def decompose(self):
...
+
+        argvals = [np.array([argval(a) for a in arg]) if is_any_list(arg)
+                   else argval(arg) for arg in self.args]
```
Comment (tias) [inline cpmpy/expressions/globalconstraints.py:461]: this still does not guarantee that it is a np.array (the latter case), which you need for the .all() below to work?

if it does, please add a comment that we assume the args are a numpy array here...

File: cpmpy/expressions/globalconstraints.py [hunk 3]
```diff
@@ -455,7 +455,15 @@ def decompose(self):
...
+
+        if any(a is None for a in argvals):
+            return None
+
+        start, dur, end, demand, cap = argvals
         # start and end seperated by duration
         if not (start + dur == end).all():
```
Comment (tias) [inline cpmpy/expressions/globalconstraints.py:468]: an alternative fix could be not to assume that it is an nparray...

e.g. `if all(s + d == e for s,d,e in zip(start, dur, end))` or similar?

File: tests/test_constraints.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 import pytest
 
-SOLVERNAME = None
+SOLVERNAME = "ortools"
```
Comment (tias) [inline tests/test_constraints.py:6]: I've done that in a separate PR with additional comments, so need not be here

File: cpmpy/expressions/globalconstraints.py (hunk 1 above)
Comment (IgnaceBleukx) [reply cpmpy/expressions/globalconstraints.py:458]: Can be on top level, will fix

File: cpmpy/expressions/globalconstraints.py (hunk 2 above)
Comment (IgnaceBleukx) [reply cpmpy/expressions/globalconstraints.py:461]: start, end and duration will always be lists (as fixed by Helge's pull request), so non-numpied args here are only the capacity and potentially the demand if fixed for all tasks.

File: cpmpy/expressions/globalconstraints.py (hunk 3 above)
Comment (IgnaceBleukx) [reply cpmpy/expressions/globalconstraints.py:468]: Also possible, but non-numpiefied decomposition of the capacity limit is much harder imho

**Comment by IgnaceBleukx** [comment]
Handled the initialization with some more care

**Comment by IgnaceBleukx** [comment]
Added a testcase. Oke to merge @tias ?
